core/wiki/macros/toolbar.tid

```diff
@@ -0,0 +1,73 @@
+title: $:/core/macros/toolbar
+tags: $:/tags/Macro
+
+\define toolbar-filter-display(tag,config)
+<$set name="config" filter="[<__config__>]match[]]" 
+		value="[all[shadows+tiddlers]tag[$tag$]!has[draft.of]]" 
+		emptyValue="[all[shadows+tiddlers]tag[$tag$]!has[draft.of]] :filter[lookup:show[$config$]!match[hide]]" >
+	<<config>>
+</$set>
+\end
+
+\define basic-buttons(tag,config,class)
+\whitespace trim
+<$list filter=<<toolbar-filter-display "$tag$" "$config$">> variable="listItem">
+	<$set name="tv-config-toolbar-class" filter="[<tv-config-toolbar-class>] [<listItem>encodeuricomponent[]addprefix[tc-btn-]] [<__class__>]" >
+		<$transclude tiddler=<<listItem>>/>
+	</$set>
+</$list>
+\end
+
+\define buttons-with-template(tag,config,class,template)
+\whitespace trim
+<$list filter=<<toolbar-filter-display "$tag$" "$config$">> >
+	<$set name="tv-config-toolbar-class" filter="[<tv-config-toolbar-class>] [<listItem>encodeuricomponent[]addprefix[tc-btn-]] [<__class__>]" >
+		<$transclude tiddler="$template$" />
+	</$set>
+</$list>
+\end
+
+\define toolbar(tag,config,class,template)
+\whitespace trim
+<$set name="buttons" 
+		filter="[<__template__>!match[]]" 
+		value="""<$macrocall $name="buttons-with-template" tag=<<__tag__>> config= <<__config__>> class=<<__class__>> template=<<__template__>> />""" 
+		emptyValue="""<$macrocall $name="basic-buttons" tag=<<__tag__>> config= <<__config__>> class=<<__class__>> />""" >
+	<<buttons>>
+</$set>
+\end
+
+
+\define toolbar-filter-drop-down(tag,config)
+[all[shadows+tiddlers]tag[$tag$]!has[draft.of]] :filter[lookup:show[$config$]match[hide]]
+\end
+
+\define toolbar-basic-drop-down-list(tag,config,class)
+<$list filter=<<toolbar-filter-drop-down "$tag$" "$config$">> variable="listItem">
+	<$set name="tv-config-toolbar-class" filter="[<tv-config-toolbar-class>] [<listItem>encodeuricomponent[]addprefix[tc-btn-]] [<__class__>]" >
+		<$transclude tiddler=<<listItem>>/>
+	</$set>
+</$list>
+\end
+
+\define toolbar-drop-down-list-wt(tag,config,class,template)
+<$list filter=<<toolbar-filter-drop-down "$tag$" "$config$">> >
+    <$set name="tv-config-toolbar-class" filter="[<tv-config-toolbar-class>] [<listItem>encodeuricomponent[]addprefix[tc-btn-]] [<__class__>]" >
+		<$transclude tiddler="$template$" />
+	</$set>
+</$list>
+\end
+
+\define toolbar-drop-down-list(tag,config,class,template)
+\whitespace trim
+<$vars tv-config-toolbar-icons="yes" 
+		tv-config-toolbar-text="yes" 
+		tv-config-toolbar-class="tc-btn-invisible">
+    <$set name="drop-down" 
+			filter=" [<__template__>!match[]]" 
+			value="""<$macrocall $name="toolbar-drop-down-list-wt" tag=<<__tag__>> config= <<__config__>> class=<<__class__>> template=<<__template__>> />""" 
```

Comment:
This line of code seems to be a duplicate ... `config= <<__config__>>`  ... Is it possible to use a macro here?

Reply:
Yes, it is reused code (and typos...).

I think that it is possible but I also think that it may be complex and I don't know how many other times would be used. It would be a macro to differenciate other two macros by one(¿two, three, ...?) parameter(s) and to choose one macro if you call to one (¿two, three, ...?) differentiator parameter(s) It looks like a overkill for me.

Reply:
OK.